test/ServerSpec.hs

```diff
@@ -0,0 +1,112 @@
+                                                                                                                , userBestFriend = Nothing
+                                                                                                                , userFriends = Nothing }
+                                                                                                         )
+                                                                                    }
+                                                                             )
+                                                        }
+                                             )
+                              }
+                         )
+
+getUserTestString :: ByteString
+getUserTestString = " query { \n getUser(id: 1) { \n userId \n userName \n userBestFriend { \n userName \n } \n } \n } \n "
+
+getUserTestQuery :: Either String (Gql GqlQuery () () 'Query)
+getUserTestQuery = Right (Gql { query = Just (ANil
+                                             , GqlQuery { getAllUsers = Nothing
+                                                        , getUser = Just (Arg (Id 1) :@@ ANil
+                                                                             , User { userId = Just (Arg Nothing :@@ ANil ,())
+                                                                                    , userName = Just (ANil ,())
+                                                                                    , userBestFriend = Just (Arg Nothing :@@ ANil
+                                                                                                            , User { userId = Nothing
+                                                                                                                   , userName = Just (ANil ,())
+                                                                                                                   , userBestFriend = Nothing
+                                                                                                                   , userFriends = Nothing 
+                                                                                                                   }
+                                                                                                            )
+                                                                                    , userFriends = Nothing
+                                                                                    }
+                                                                             )
+                                                        }
+                                             )
+                              }
+                         )
+
+
+deriving instance Show (GqlQuery 'Response)
+deriving instance ToJSON (GqlQuery 'Response)
+deriving instance Show (GqlQuery 'Query)
+
+instance Arbitrary (User 'Query) where
+  arbitrary = recordGen
```

Comment:
I learned the other day that if you also add `shrink = genericShrink` here, counterexamples will come out much more nicely :)